### ISSUE TITLE
Drop accuracy from 0.16 to 10^-5 at learning.

### ISSUE THREAD

**Issue by igriffer** [BVLC/caffe#2436]
I have 1000 ImageNet classes and 1320 own classes of objects.
In previous time, 1000 ImageNet classes plus my 4 classes learned perfectly.
It is very strange and I do not know, is this a problem of a sample base or a bug.

Nvidia Tesla k20c, Cuda 6.5, CUDNN, Ubuntu 14.04

Learning log:
```
I0509 19:45:41.120270 20669 solver.cpp:266] Iteration 7000, Testing net (#0)
I0509 19:47:26.425755 20669 solver.cpp:315]     Test net output #0: accuracy = 0.16014
I0509 19:47:26.425858 20669 solver.cpp:315]     Test net output #1: loss = 5.5315 (* 1 = 5.5315 loss)
I0509 19:47:26.917171 20669 solver.cpp:189] Iteration 7000, loss = 4.15675
I0509 19:47:26.917219 20669 solver.cpp:204]     Train net output #0: loss = 4.15675 (* 1 = 4.15675 loss)
I0509 19:47:26.917232 20669 solver.cpp:464] Iteration 7000, lr = 0.01
I0509 19:49:46.336627 20669 solver.cpp:189] Iteration 7100, loss = 4.38882
I0509 19:49:46.336810 20669 solver.cpp:204]     Train net output #0: loss = 4.38882 (* 1 = 4.38882 loss)
I0509 19:49:46.336827 20669 solver.cpp:464] Iteration 7100, lr = 0.01
I0509 19:52:05.722803 20669 solver.cpp:189] Iteration 7200, loss = 4.33491
I0509 19:52:05.722961 20669 solver.cpp:204]     Train net output #0: loss = 4.33491 (* 1 = 4.33491 loss)

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 3000 iter ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

I0509 21:00:37.677863 20669 solver.cpp:334] Snapshotting to /media/3tb/example_mix/model/_iter_10000.caffemodel
I0509 21:00:38.959280 20669 solver.cpp:342] Snapshotting solver state to /media/3tb/example_mix/model/_iter_10000.solverstate
I0509 21:00:39.645733 20669 solver.cpp:266] Iteration 10000, Testing net (#0)
I0509 21:02:24.072829 20669 solver.cpp:315]     Test net output #0: accuracy = 0.0359002
I0509 21:02:24.072985 20669 solver.cpp:315]     Test net output #1: loss = 6.72778 (* 1 = 6.72778 loss)
I0509 21:02:24.565165 20669 solver.cpp:189] Iteration 10000, loss = 6.11271
I0509 21:02:24.565210 20669 solver.cpp:204]     Train net output #0: loss = 6.11271 (* 1 = 6.11271 loss)
I0509 21:02:24.565223 20669 solver.cpp:464] Iteration 10000, lr = 0.01
I0509 21:04:43.930517 20669 solver.cpp:189] Iteration 10100, loss = 5.76941
I0509 21:04:43.930681 20669 solver.cpp:204]     Train net output #0: loss = 5.76941 (* 1 = 5.76941 loss)
I0509 21:04:43.930701 20669 solver.cpp:464] Iteration 10100, lr = 0.01
I0509 21:07:03.297937 20669 solver.cpp:189] Iteration 10200, loss = 5.95765
I0509 21:07:03.298082 20669 solver.cpp:204]     Train net output #0: loss = 5.95765 (* 1 = 5.95765 loss)
I0509 21:07:03.298100 20669 solver.cpp:464] Iteration 10200, lr = 0.01
I0509 21:09:22.695098 20669 solver.cpp:189] Iteration 10300, loss = 6.87054
I0509 21:09:22.695286 20669 solver.cpp:204]     Train net output #0: loss = 6.87054 (* 1 = 6.87054 loss)
I0509 21:09:22.695303 20669 solver.cpp:464] Iteration 10300, lr = 0.01
I0509 21:11:42.033999 20669 solver.cpp:189] Iteration 10400, loss = 6.00503
I0509 21:11:42.034142 20669 solver.cpp:204]     Train net output #0: loss = 6.00503 (* 1 = 6.00503 loss)
I0509 21:11:42.034164 20669 solver.cpp:464] Iteration 10400, lr = 0.01
I0509 21:14:01.349205 20669 solver.cpp:189] Iteration 10500, loss = 87.3365
I0509 21:14:01.349460 20669 solver.cpp:204]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0509 21:14:01.349477 20669 solver.cpp:464] Iteration 10500, lr = 0.01
I0509 21:16:20.426568 20669 solver.cpp:189] Iteration 10600, loss = 87.3365
I0509 21:16:20.426709 20669 solver.cpp:204]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0509 21:16:20.426726 20669 solver.cpp:464] Iteration 10600, lr = 0.01
I0509 21:18:39.481897 20669 solver.cpp:189] Iteration 10700, loss = 87.3365
I0509 21:18:39.482095 20669 solver.cpp:204]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0509 21:18:39.482137 20669 solver.cpp:464] Iteration 10700, lr = 0.01
I0509 21:20:58.570801 20669 solver.cpp:189] Iteration 10800, loss = 87.3365
I0509 21:20:58.570925 20669 solver.cpp:204]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0509 21:20:58.570948 20669 solver.cpp:464] Iteration 10800, lr = 0.01
I0509 21:23:17.652310 20669 solver.cpp:189] Iteration 10900, loss = 87.3365
I0509 21:23:17.652403 20669 solver.cpp:204]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0509 21:23:17.652418 20669 solver.cpp:464] Iteration 10900, lr = 0.01
I0509 21:25:35.318344 20669 solver.cpp:266] Iteration 11000, Testing net (#0)
I0509 21:27:21.126389 20669 solver.cpp:315]     Test net output #0: accuracy = 8e-05
I0509 21:27:21.126502 20669 solver.cpp:315]     Test net output #1: loss = 87.3361 (* 1 = 87.3361 loss)
I0509 21:27:21.617015 20669 solver.cpp:189] Iteration 11000, loss = 87.3365
I0509 21:27:21.617064 20669 solver.cpp:204]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0509 21:27:21.617077 20669 solver.cpp:464] Iteration 11000, lr = 0.01
I0509 21:29:40.728096 20669 solver.cpp:189] Iteration 11100, loss = 87.3365
I0509 21:29:40.728183 20669 solver.cpp:204]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0509 21:29:40.728199 20669 solver.cpp:464] Iteration 11100, lr = 0.01
I0509 21:31:59.799192 20669 solver.cpp:189] Iteration 11200, loss = 87.3365
I0509 21:31:59.799352 20669 solver.cpp:204]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0509 21:31:59.799383 20669 solver.cpp:464] Iteration 11200, lr = 0.01
I0509 21:34:18.862293 20669 solver.cpp:189] Iteration 11300, loss = 87.3365
I0509 21:34:18.862431 20669 solver.cpp:204]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0509 21:34:18.862447 20669 solver.cpp:464] Iteration 11300, lr = 0.01
I0509 21:36:37.953515 20669 solver.cpp:189] Iteration 11400, loss = 87.3365
I0509 21:36:37.953608 20669 solver.cpp:204]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0509 21:36:37.953624 20669 solver.cpp:464] Iteration 11400, lr = 0.01
I0509 21:38:57.053948 20669 solver.cpp:189] Iteration 11500, loss = 87.3365
I0509 21:38:57.054029 20669 solver.cpp:204]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0509 21:38:57.054044 20669 solver.cpp:464] Iteration 11500, lr = 0.01
I0509 21:41:16.101441 20669 solver.cpp:189] Iteration 11600, loss = 87.3365
I0509 21:41:16.101601 20669 solver.cpp:204]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0509 21:41:16.101624 20669 solver.cpp:464] Iteration 11600, lr = 0.01
I0509 21:43:35.153794 20669 solver.cpp:189] Iteration 11700, loss = 87.3365
I0509 21:43:35.153937 20669 solver.cpp:204]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0509 21:43:35.153954 20669 solver.cpp:464] Iteration 11700, lr = 0.01
I0509 21:45:54.248154 20669 solver.cpp:189] Iteration 11800, loss = 87.3365
I0509 21:45:54.248296 20669 solver.cpp:204]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0509 21:45:54.248313 20669 solver.cpp:464] Iteration 11800, lr = 0.01
I0509 21:48:13.350260 20669 solver.cpp:189] Iteration 11900, loss = 87.3365
I0509 21:48:13.354642 20669 solver.cpp:204]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0509 21:48:13.354660 20669 solver.cpp:464] Iteration 11900, lr = 0.01
I0509 21:50:31.060969 20669 solver.cpp:266] Iteration 12000, Testing net (#0)
I0509 21:52:17.197026 20669 solver.cpp:315]     Test net output #0: accuracy = 6e-05
I0509 21:52:17.197129 20669 solver.cpp:315]     Test net output #1: loss = 87.3361 (* 1 = 87.3361 loss)
```

Solver:
```
net: "caffe/mixed/train_val.prototxt"
test_iter: 1000
test_interval: 1000
base_lr: 0.01
lr_policy: "step"
gamma: 0.1
stepsize: 70000
display: 100
max_iter: 450000
momentum: 0.9
weight_decay: 0.0005
snapshot: 5000
snapshot_prefix: "/media/3tb/example_mix/model/"
solver_mode: GPU
```

train_val:
```
name: "CaffeNet"
layer {
  name: "data"
  type: "Data"
  top: "data"
  top: "label"
  include {
    phase: TRAIN
  }
  transform_param {
    mirror: true
    crop_size: 227
    mean_file: "caffe_porn/imagenet_mean.binaryproto"
  }
# mean pixel / channel-wise mean instead of mean image
#  transform_param {
#    crop_size: 227
#    mean_value: 104
#    mean_value: 117
#    mean_value: 123
#    mirror: true
#  }
  data_param {
    source: "/media/3tb/example_mix/ilsvrc12_train_lmdb"
    batch_size: 256
    backend: LMDB
  }
}
layer {
  name: "data"
  type: "Data"
  top: "data"
  top: "label"
  include {
    phase: TEST
  }
  transform_param {
    mirror: false
    crop_size: 227
    mean_file: "caffe_porn/imagenet_mean.binaryproto"
  }
# mean pixel / channel-wise mean instead of mean image
#  transform_param {
#    crop_size: 227
#    mean_value: 104
#    mean_value: 117
#    mean_value: 123
#    mirror: true
#  }
  data_param {
    source: "/media/3tb/example_mix/ilsvrc12_val_lmdb"
    batch_size: 50
    backend: LMDB
  }
}
layer {
  name: "conv1"
  type: "Convolution"
  bottom: "data"
  top: "conv1"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 96
    kernel_size: 11
    stride: 4
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "relu1"
  type: "ReLU"
  bottom: "conv1"
  top: "conv1"
}
layer {
  name: "pool1"
  type: "Pooling"
  bottom: "conv1"
  top: "pool1"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "norm1"
  type: "LRN"
  bottom: "pool1"
  top: "norm1"
  lrn_param {
    local_size: 5
    alpha: 0.0001
    beta: 0.75
  }
}
layer {
  name: "conv2"
  type: "Convolution"
  bottom: "norm1"
  top: "conv2"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 256
    pad: 2
    kernel_size: 5
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu2"
  type: "ReLU"
  bottom: "conv2"
  top: "conv2"
}
layer {
  name: "pool2"
  type: "Pooling"
  bottom: "conv2"
  top: "pool2"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "norm2"
  type: "LRN"
  bottom: "pool2"
  top: "norm2"
  lrn_param {
    local_size: 5
    alpha: 0.0001
    beta: 0.75
  }
}
layer {
  name: "conv3"
  type: "Convolution"
  bottom: "norm2"
  top: "conv3"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 384
    pad: 1
    kernel_size: 3
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "relu3"
  type: "ReLU"
  bottom: "conv3"
  top: "conv3"
}
layer {
  name: "conv4"
  type: "Convolution"
  bottom: "conv3"
  top: "conv4"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 384
    pad: 1
    kernel_size: 3
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu4"
  type: "ReLU"
  bottom: "conv4"
  top: "conv4"
}
layer {
  name: "conv5"
  type: "Convolution"
  bottom: "conv4"
  top: "conv5"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 256
    pad: 1
    kernel_size: 3
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu5"
  type: "ReLU"
  bottom: "conv5"
  top: "conv5"
}
layer {
  name: "pool5"
  type: "Pooling"
  bottom: "conv5"
  top: "pool5"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "fc6"
  type: "InnerProduct"
  bottom: "pool5"
  top: "fc6"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 4096
    weight_filler {
      type: "gaussian"
      std: 0.005
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu6"
  type: "ReLU"
  bottom: "fc6"
  top: "fc6"
}
layer {
  name: "drop6"
  type: "Dropout"
  bottom: "fc6"
  top: "fc6"
  dropout_param {
    dropout_ratio: 0.5
  }
}
layer {
  name: "fc7"
  type: "InnerProduct"
  bottom: "fc6"
  top: "fc7"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 6960
    weight_filler {
      type: "gaussian"
      std: 0.005
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu7"
  type: "ReLU"
  bottom: "fc7"
  top: "fc7"
}
layer {
  name: "drop7"
  type: "Dropout"
  bottom: "fc7"
  top: "fc7"
  dropout_param {
    dropout_ratio: 0.5
  }
}
layer {
  name: "fc8"
  type: "InnerProduct"
  bottom: "fc7"
  top: "fc8"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 2320
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "accuracy"
  type: "Accuracy"
  bottom: "fc8"
  bottom: "label"
  top: "accuracy"
  include {
    phase: TEST
  }
}
layer {
  name: "loss"
  type: "SoftmaxWithLoss"
  bottom: "fc8"
  bottom: "label"
  top: "loss"
}
```

**Comment by n-zhang** [comment]
I suspect the learning diverges and the gradient or activation becomes nan, you can turn on the debug flag to check. Please ask usage questions on the caffe-users list. Thanks!
